xla/service/gpu/gpu_compiler.cc

```diff
@@ -858,8 +858,6 @@ absl::Status GpuCompiler::OptimizeHloModule(
       pipeline.AddPass<AlgebraicSimplifier>(layout_insensitive_algsimp_opts);
     }();
 
-    pipeline.AddPass<HloComputationDeduplicator>(
-        /*mark_fusion_duplications=*/false);
```

Comment:
Is this intended as part of this PR?

Reply:
Yes, this will dedup comps incorrectly, at least without [this](https://github.com/openxla/xla/pull/8800) it will.
We call flattenCallGraph multiple times anyway so this call is redundant.

Reply:
That still seems like a separate change to me, which should thus have its own separate PR: is it possible to split this out?

Reply:
Without this, some scenario will continue to fail since comps are still incorrectly comped. We can wait for the above fix to land then we can either leave this as is or remove it in another pr.

Reply:
How about making float normalization itself duplicate the computation if it is shared between a collective and non-collective instruction? I have sketched that approach in a draft PR: https://github.com/openxla/xla/pull/8896

With that approach, float normalization will not rely on any sharing invariants.

What do you think?

Reply:
I'm ok with either approach. One question I had is: I think we all agreed that collective comps cannot be shared with other ops, so if we actually see this case, we should just error out instead of trying to work around an illegal IR.
IMO, the correct flow should be that HloVerifier takes the responsibility to check for illegal IR, and then float norm pass just does a local analysis to skip type promotion.